### PR TITLE
Add support for Traceability Matrix Tables

### LINKED ISSUE #61: CTAL-TA Traceability matrix table in LaTeX
Despite the provided code in #51 [comment](https://github.com/istqborg/istqb_product_base/issues/51#issuecomment-2145716482) I am not able to pull the traceability matrix for CTAL TA and I am not able to find an issue with the code.

Initially I have only tried the [first part with BOs](https://github.com/istqborg/istqb-ctal-ta/commit/a7801f604d824a5a404e5a9185e5de0ecbd50d37), but the pipeline failed and cannot find out the problem.

Later I have [added all LOs](https://github.com/istqborg/istqb-ctal-ta/commit/18e3d5fd285231e73d9d29e0ec4f59a3342c6886), which makes the full table, but of course, with the same result.

Can you please:

1. Find the issue(s) I made
2. Suggest how to debug pipeline failures for LaTeX tables.
3. Suggest how to debug LaTeX tables creation. Is it possible to use some online tooling? Or syntax checker? Overleaf didn't worked for me :(
4. Add these hints to the example document.

### PR DESCRIPTION
**Pull request by Witiko**
This PR makes the following changes:

- [x] Collect and store Learning Objectives in data structures, see the demonstration in [\#61 (comment)](https://github.com/istqborg/istqb_product_base/issues/61#issuecomment-2265483685).
- [x] Add markup for Business Outcomes to example document.
- [x] Collect and store Business Outcomes in data structures.
- [x] Add markup for Business Outcomes to other repositories that currently contain business outcomes:
    - https://github.com/istqborg/istqb-ct-qdo/pull/30
    - https://github.com/istqborg/istqb-ctal-ta/pull/10
    - https://github.com/istqborg/istqb-ct-atlas/pull/2
    - https://github.com/istqborg/istqb-ct-tas/pull/7
    - https://github.com/istqborg/istqb-ct-tae/pull/9
    - https://github.com/istqborg/istqb_product_template/pull/17
    - https://github.com/istqborg/istqb-ctfl/pull/18
- [x] Translate headers "Code" and "Name" for Business Outcomes, see e.g. document `syllabus-de.tex` from `istqb-ct-tas`.
- [x] Add a schema for files `traceability-matrix.yml` that specify a Traceability Matrix between Learning Objectives and Business Outcomes and add an example file `traceability-matrix.yml` for the example document.
- [x] Add snippet `istqb/common/traceability-matrix` that processes `traceability-matrix.yml`.
- [x] Update snippet `istqb/common/traceability-matrix`, so that it also typesets an appendix with a Traceability Matrix Table.
- [x] Update snippet `istqb/common/traceability-matrix`, so that it also typesets the full Traceability Matrix Table.
- [x] Demonstrate the snippet `istqb/common/traceability-matrix` on the example document.

This PR also makes the following changes:

- Add argument `--preview-continuous` to command `compile-tex-to-pdf`:
  ```
  usage: template.py compile-tex-to-pdf [-h] [-pvc]
  
  options:
    -h, --help            show this help message and exit
    -pvc, --previous-continuous
                          Keep recompiling the TeX files as they change
  ```
  This enables the interactive continuous compilation of TeX files.

- Support partial recompilation with the `compile-tex-to-pdf` command.
- Remove tabs from the source code.

Closes #61.

### CODE REVIEW

File: example-document.tex [hunk 1]
```diff
@@ -44,4 +44,53 @@
 % Index
 \printindex
 
+% Learning objectives
```
Comment (danopolan) [inline example-document.tex:47]: Do you know if this section will be necessary after finishing all tasks within PR #90? 
This seems quite extensive and could be scary for the users without programming backround.

File: template/markdownthemeistqb_common.sty [hunk 2]
```diff
@@ -602,6 +605,66 @@
...
+                                          \g_istqb_business_outcomes_numbers_seq
+                                          { ########1 }
+                                        \prop_gput:Nnn
+                                          \g_istqb_business_outcomes_names_prop
+                                          { ########1 }
+                                          { ################1 }
+                                        \tl_put_right:Nn
+                                          \l_istqb_business_outcomes_table_tl
+                                          {
+                                            \g_istqb_prefix_tl - ########1 &
```
Comment (danopolan) [inline template/markdownthemeistqb_common.sty:642]: BO code should be set as `prefix-BO1`. So I assume this should be 

```suggestion
                                            \g_istqb_prefix_tl - BO########1 &
```

**Review by danopolan — APPROVED** [review]

**Review by danopolan — CHANGES_REQUESTED** [review]
good progress so far

File: example-document.tex (hunk 1 above)
Comment (Witiko) [reply example-document.tex]: The section was only meant for debugging, it must have slipped in with the other changes. I fixed this in commit 727ed35 from this PR.

**Comment by Witiko** [comment]
@danopolan For your review, please see my demo in https://github.com/istqborg/istqb_product_base/issues/61#issuecomment-2301793836.

**Review by danopolan — APPROVED** [review]

**Pull request merged**